src/diff.rs

```diff
@@ -765,6 +901,80 @@ impl Drop for DiffStats {
     }
 }
 
+impl<'a> DiffBinary<'a> {
+    /// The contents of the old file.
+    pub fn old_file(&self) -> DiffBinaryFile<'a> {
+        unsafe { Binding::from_raw(&(*self.raw).old_file as *const _) }
+    }
+
+    /// The contents of the new file.
+    pub fn new_file(&self) -> DiffBinaryFile<'a> {
+        unsafe { Binding::from_raw(&(*self.raw).new_file as *const _) }
+    }
+}
+
+impl<'a> Binding for DiffBinary<'a> {
+    type Raw = *const raw::git_diff_binary;
+    unsafe fn from_raw(raw: *const raw::git_diff_binary) -> DiffBinary<'a> {
+        DiffBinary {
+            raw: raw,
+            _marker: marker::PhantomData,
+        }
+    }
+    fn raw(&self) -> *const raw::git_diff_binary { self.raw }
+}
+
+impl<'a> DiffBinaryFile<'a> {
+    /// The type of binary data for this file
+    pub fn kind(&self) -> DiffBinaryKind {
+        unsafe { Binding::from_raw((*self.raw).kind) }
+    }
+
+    /// The binary data, deflated
+    pub fn data(&self) -> &[u8] {
+        unsafe {
+            slice::from_raw_parts((*self.raw).data as *const u8,
+                                  (*self.raw).datalen as usize)
+        }
+    }
+
+    /// The length of the binary data after inflation
+    pub fn inflated_len(&self) -> usize {
+        unsafe { (*self.raw).inflatedlen as usize }
+    }
+
+}
+
+impl<'a> Binding for DiffBinaryFile<'a> {
+    type Raw = *const raw::git_diff_binary_file;
+    unsafe fn from_raw(raw: *const raw::git_diff_binary_file) -> DiffBinaryFile<'a> {
+        DiffBinaryFile {
+            raw: raw,
+            _marker: marker::PhantomData,
+        }
+    }
+    fn raw(&self) -> *const raw::git_diff_binary_file { self.raw }
+}
+
+impl Binding for DiffBinaryKind {
+    type Raw = raw::git_diff_binary_t;
+    unsafe fn from_raw(raw: raw::git_diff_binary_t) -> DiffBinaryKind {
+        match raw {
+            raw::GIT_DIFF_BINARY_NONE => DiffBinaryKind::None,
+            raw::GIT_DIFF_BINARY_LITERAL => DiffBinaryKind::Literal,
+            raw::GIT_DIFF_BINARY_DELTA => DiffBinaryKind::Delta,
+            _ => panic!("Unknown git diff binary kind"),
+        }
+    }
+    fn raw(&self) -> raw::git_diff_binary_t {
+        match self {
```

Comment:
stylistically this is typically done via `match *self` and then you can avoid the leading `&` in front of each arm